internal/monitors/vsphere/model/model.go

```diff
@@ -0,0 +1,100 @@
+package model
+
+import (
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	// "real-time" vsphereInfo metrics are available in 20 second intervals
+	VSMetricsInterval = 20
+	// ten minutes
+	DefaultVSInfoReloadInterval = 600
+)
+
+const (
+	DatacenterType = "Datacenter"
+	ClusterType    = "ClusterComputeResource"
+	VMType         = "VirtualMachine"
+	HostType       = "HostSystem"
+)
+
+const objectType = "object_type"
+
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+	Host                 string `yaml:"host"`
+	Port                 uint16 `yaml:"port"`
+	// The vSphere username.
+	Username string `yaml:"username"`
+	// The vSphere password.
+	Password string `yaml:"password"`
+	// Whether we verify the server's certificate chain and host name
+	InsecureSkipVerify bool `yaml:"InsecureSkipVerify"`
+	// How often to reload the inventory and inventory metrics.
+	InventoryRefreshIntervalSeconds int `yaml:"inventoryRefreshIntervalSeconds"`
```

Comment:
You can make this a `time.Duration` and set the default to be "10s" and it'll magically convert it. Example: https://github.com/signalfx/signalfx-agent/blob/e3e5a713f34ffc053b447f6da9aa939be2f0a09e/internal/monitors/logstash/tcp/monitor.go#L48

Reply:
Nice, I'll do that instead.

Reply:
Be sure to take the suffix "Seconds" out if you do.

Reply:
Done.